File: orientedDisign/js_demo/userFactory.js
const { Student, Teacher } = require('./user');

class UserFactory {
  static createUser(type, userId, name, email) {
    const t = (type || '').toLowerCase();
    if (t === 'student') return new Student(userId, name, email);
    if (t === 'teacher') return new Teacher(userId, name, email);
    throw new Error(`Unknown user type: ${type}`);
  }
}

module.exports = UserFactory;

File: orientedDisign/library/user_factory.py
from library.user import Student, Teacher, User


class UserFactory:
    @staticmethod
    def create_user(user_type: str, user_id: str, name: str, email: str) -> User:
        t = user_type.lower()
        if t == "student":
            return Student(user_id, name, email)
        if t == "teacher":
            return Teacher(user_id, name, email)
        raise ValueError(f"Unknown user type: {user_type}")
